pkg/asset/agent/installconfig.go

```diff
@@ -82,6 +86,26 @@ func (a *OptionalInstallConfig) validateSupportedPlatforms(installConfig *types.
 	return allErrs
 }
 
+func (a *OptionalInstallConfig) validateSupportedArchs(installConfig *types.InstallConfig) field.ErrorList {
+	var allErrs field.ErrorList
+
+	fieldPath := field.NewPath("ControlPlane", "Architecture")
+
+	if string(installConfig.ControlPlane.Architecture) != types.ArchitectureAMD64 {
+		allErrs = append(allErrs, field.NotSupported(fieldPath, installConfig.ControlPlane.Architecture, []string{"x86-64", "x86_64", types.ArchitectureAMD64}))
+	}
+
+	for i, compute := range installConfig.Compute {
+		fieldPath := field.NewPath(fmt.Sprintf("Compute[%d]", i), "Architecture")
+
+		if string(compute.Architecture) != types.ArchitectureAMD64 {
+			allErrs = append(allErrs, field.NotSupported(fieldPath, compute.Architecture, []string{"x86-64", "x86_64", types.ArchitectureAMD64}))
```

Comment:
Let's advertise only amd64 as supported, regardless of whether we translate internally.